### ISSUE TITLE
Any way to only submit values that are present in the form?

### ISSUE THREAD

**Issue by jomag** [jaredpalmer/formik#515]
Question/Feature

I have a redux store that keeps a couple of hundred settings. Of those I only want to display 10-20 in 
the form. No problem so far. But when I submit the form, *all* values are submitted, not only those present in the form (the values with an input field).

I suppose I could pick just the values I need from the store state in `mapPropsToValues`, but it seems like duplicated work.

Is there any better way to handle this?

**Comment by latviancoder** [comment]
Jep, Formik has no idea about your DOM, it only works with supplied values. So if your form is dynamic and you don't want unnecessary values to be submitted/touched, then you also have to supply them dynamically. Here is an example from a dummy app I am currently building:

```
<Checkbox name="hasOtherFriend" onChange={isChecked => {
    setValues({
        ...values,
        otherFriend: isChecked ? {name: '', age: ''} : undefined
    });
}}/>
```

This doesn't feel right, but I don't know any other solution. It would be of course cool if `mapsPropsToValues` could somehow work the same way `validationSchema` does (using `when`):

```
otherFriend: object().when('hasOtherFriend', {
    is: true,
    then: friendValidationSchema
}),
```

@jaredpalmer what are your thoughts on this? Is there a better approach for dynamic forms?

**Comment by jaredpalmer** [comment]
Yeah this is still hard unfortunately but `setValues` is the move. Curious how you would do this without Formik (i.e. with  just vanilla react)?

**Comment by klis87** [comment]
@jomag If you wanna limit fields, you can use `yup` for this. I do it for highly dynamic forms, using `when` like @latviancoder mentioned, but also somehow I abuse `yup.validate` with `stripUnknown: true` - https://github.com/jquense/yup#mixedvalidatevalue-any-options-object-promiseany-validationerror

Another trick I use is `const initialValues =  schema.cast()`, which can give me default values of a `schema`. 

My point is, for me `yup` is a logic/validation layer, `Formik` is a view layer, each one has its own responsibility.

**Comment by latviancoder** [comment]
@klis87 how are you using schema.cast() exactly? It requires at least 1 parameter.

@jaredpalmer no idea honestly, luckily for me I haven't worked on such a project yet.

**Comment by klis87** [comment]
@latviancoder it doesnt, try it :) Maybe this is not intended, but for example it will give you default values set in your schema.

**Comment by hrgui** [comment]
I'd assume this is prevalent on edit, in which I do a deep diff between the initial set of values and the values I am about to send, then I send the values.

However, the diff is too good that it deleted things like my ID, so I have to add that back manually.
